sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala

```diff
@@ -114,6 +114,15 @@ private[sql] object TypeCollection {
     BooleanType,
     ByteType, ShortType, IntegerType, LongType)
 
+  /**
+   * Types that include numeric types and interval type. They are only used in unary_minus,
+   * unary_positive, add and subtract operations.
+   */
+  val NumericAndInterval = TypeCollection(
+    ByteType, ShortType, IntegerType, LongType,
+    FloatType, DoubleType, DecimalType,
+    IntervalType)
```

Comment:
I saw there are already use cases for directly showing `numeric` in error message. So I updated this too.